include/SFML/System/Clock.hpp

```diff
@@ -69,13 +57,14 @@ namespace priv
 ///
 ////////////////////////////////////////////////////////////
 #if defined(SFML_SYSTEM_ANDROID) && defined(SFML_ANDROID_USE_SUSPEND_AWARE_CLOCK)
-using MostSuitableClock = SuspendAwareClock;
+using ClockImpl = SuspendAwareClock;
 #else
-using MostSuitableClock = std::conditional_t<std::chrono::high_resolution_clock::is_steady,
-                                             std::chrono::high_resolution_clock,
-                                             std::chrono::steady_clock>;
+using ClockImpl = std::chrono::steady_clock;
```

Comment:
https://en.cppreference.com/w/cpp/chrono/high_resolution_clock

<img width="808" alt="Screenshot 2023-02-08 at 12 04 03 AM" src="https://user-images.githubusercontent.com/39244355/217458105-f869ee44-cadb-41d2-ab92-a365384aff7c.png">

When a high resolution clock is steady, it's just an alias for `steady_clock`. We can avoid this template logic by directly using `steady_clock`.

Reply:
Is often != is always. I believe we had quite some discussion about this topic when it was implemented, so I don't really feel comfortable just ignoring that because of a note on cppreference

Reply:
See the discussion here: https://github.com/SFML/SFML/pull/1872

Reply:
I think it's a safe thing to do. We're already checking that the clock has the microsecond precision that we need. Does it really matter if there's a hypothetical nanosecond resolution steady clock (via `high_resolution_clock`) if `sf::Clock` doesn't even expose that level of precision? I'm not sure how anyone could take advantage of that precision since it'll get truncated to microseconds.

Reply:
I think it does matter. The same question can be asked, does it really matter to have two additional lines of code to get the best matching clock?

Reply:
Pretend that `steady_clock` is steady with a precision of microseconds and `high_resolution_clock` is steady with a precision of nanoseconds. Given the `sf::Clock` and `sf::Time` API, how is a user supposed to take advantage of that higher precision? The API makes no mention of nanoseconds so that extra precision would get lost before the user would do anything with it.

Reply:
What is then the motivation behind this change?

It is not more correct now, arguably _less_ correct even, and we save a bit of code that was found to be the best solution after in-depth research and testing in #1872.

In particular reconndev mentioned [in this comment](https://github.com/SFML/SFML/pull/1872#issuecomment-988176144):

> In conclusion to my research, I believe to have carefully tested and checked (almost) all supported platforms and major compiler implementations of std::chrono clocks. The results favor `steady_clock` but a more general solution was provided (taking `high_resolution_clock` into account if the clock is monotonic).

Reply:
> What is then the motivation behind this change?

Removing unnecessary code. We do this all the time.

> It is not more correct now, arguably less correct even

I disagree. See my previous statement about how this can't be less correct. I'm curious what you think of that argument.

> that was found to be the best solution after in-depth research and testing in https://github.com/SFML/SFML/pull/1872.

Did someone find issues with using `steady_clock` as opposed to `high_resolution_clock`? My impression of that discussion is that we still needed `SuspendAwareClock` for Android but otherwise `steady_clock` was all we wanted. I double checked and see no evidence of platforms where `high_resolution_clock` is steady and yet also higher precision than `steady_clock`.

---

I'm in the minority so I will revert that change but I stand by it being perfectly valid and an improvement on the status quo which is unnecessary complicated. I'd love someone to explain why it's useful to use a clock with a higher precision than our API can represent. I'm not sure what that buys us even in the incredibly unlikely case it occurs.